### PR TITLE
OpenEdition queryable

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

File: contracts/eip/queryable/ERC721AQueryableUpgradeable.sol [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+     * - `extraData = <Extra data when token was burned>`
+     *
+     * Otherwise:
+     *
+     * - `addr = <Address of owner>`
+     * - `startTimestamp = <Timestamp of start of ownership>`
+     * - `burned = false`
+     * - `extraData = <Extra data at start of ownership>`
+     */
+    function explicitOwnershipOf(uint256 tokenId)
```
Comment (joaquim-verges) [inline contracts/eip/queryable/ERC721AQueryableUpgradeable.sol:51]: don't really need these. They're nice, but doesn't solve the problem at hand

File: contracts/eip/queryable/ERC721AQueryableUpgradeable.sol [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                }
+            }
+        }
+    }
+
+    /**
+     * @dev Returns an array of `TokenOwnership` structs at `tokenIds` in order.
+     * See {ERC721AQueryable-explicitOwnershipOf}
+     */
+    function explicitOwnershipsOf(uint256[] calldata tokenIds)
```
Comment (joaquim-verges) [inline contracts/eip/queryable/ERC721AQueryableUpgradeable.sol:74]: same for this

File: contracts/eip/queryable/ERC721AUpgradeable.sol [hunk 3]
```diff
@@ -0,0 +1,1111 @@
...
+    //                       OTHER OPERATIONS
+    // =============================================================
+
+    /**
+     * @dev Returns the message sender (defaults to `msg.sender`).
+     *
+     * If you are writing GSN compatible contracts, you need to override this function.
+     */
+    function _msgSenderERC721A() internal view virtual returns (address) {
+        return msg.sender;
```
Comment (joaquim-verges) [inline contracts/eip/queryable/ERC721AUpgradeable.sol:1057]: is this new? do we need to override this?

Comment (kumaryash90) [reply contracts/eip/queryable/ERC721AUpgradeable.sol:1057]: yes

updating

**Pull request merged**